packages/next/src/server/web/spec-extension/unstable-no-store.ts

```diff
@@ -1,20 +1,21 @@
-import { staticGenerationAsyncStorage } from '../../../client/components/static-generation-async-storage.external'
-import { staticGenerationBailout } from '../../../client/components/static-generation-bailout'
+import { getExpectedStaticGenerationStore } from '../../../client/components/static-generation-async-storage.external'
+import { markCurrentScopeAsDynamic } from '../../app-render/dynamic-rendering'
 
+/**
+ * Expects to be called in an App Router render and will error if not.
+ *
+ * marks the current scope as dynamic. In non PPR cases this will make a static render
+ * halt and mark the page as dynamic. In PPR cases this will postpone the render at this location.
+ *
+ * If we are inside a cache scope then this function is a noop
+ */
 export function unstable_noStore() {
-  const staticGenerationStore = staticGenerationAsyncStorage.getStore()
-
-  if (staticGenerationStore?.isUnstableCacheCallback) {
-    // if called within a next/cache call, we want to cache the result
-    // and defer to the next/cache call to handle how to cache the result.
+  const callingExpression = 'unstable_noStore()'
+  const store = getExpectedStaticGenerationStore(callingExpression)
```

Comment:
This is maybe wrong b/c calling this inside a route handler is maybe ok? (a noop) cc @wyattjoh